spec/models/metric/capture_spec.rb

```diff
@@ -1,138 +1,94 @@
 describe Metric::Capture do
-  describe ".alert_capture_threshold" do
-    let(:target) { FactoryGirl.build(:host_vmware) }
-
-    it "parses fixed num" do
-      stub_performance_settings(:capture_threshold_with_alerts => {:host => 4})
-      Timecop.freeze(Time.now.utc) do
-        expect(described_class.alert_capture_threshold(target)).to eq 4.minutes.ago.utc
-      end
-    end
-
-    it "parses string" do
-      stub_performance_settings(:capture_threshold_with_alerts => {:host => "4.minutes"})
-      Timecop.freeze(Time.now.utc) do
-        expect(described_class.alert_capture_threshold(target)).to eq 4.minutes.ago.utc
-      end
-    end
+  let(:zone) { FactoryGirl.create(:zone) }
+  let(:ems) { FactoryGirl.create(:ems_vmware) }
+  let(:ems2)  { FactoryGirl.create(:ems_vmware, :zone => zone) }
+  let(:ems3) { FactoryGirl.create(:ems_vmware, :zone => zone) }
+
+  describe ".perf_capture_timer" do
+    it "triggers better capture" do
+      EvmSpecHelper.local_miq_server.zone
+      ems  # default zone
+      ems2 # other zone
+
+      expect_queued(
+        :class_name  => "Metric::Capture",
+        :method_name => "perf_collect_all_metrics",
+        :args        => [ems.id, "realtime", nil, nil, {}],
+      )
 
-    it "produces default with class not found" do
-      stub_performance_settings(:capture_threshold_with_alerts => {:vm      => "4.minutes",
-                                                                   :default => "1.minutes"})
-      Timecop.freeze(Time.now.utc) do
-        expect(described_class.alert_capture_threshold(target)).to eq 1.minute.ago.utc
-      end
+      Metric::Capture.perf_capture_timer
     end
   end
 
-  describe ".standard_capture_threshold" do
-    let(:host) { FactoryGirl.build(:host_vmware) }
-
-    it "parses fixed num" do
-      stub_performance_settings(:capture_threshold => {:host => 4})
-      Timecop.freeze(Time.now.utc) do
-        expect(described_class.standard_capture_threshold(host)).to eq 4.minutes.ago.utc
-      end
-    end
-
-    it "parses string" do
-      stub_performance_settings(:capture_threshold => {:host => "4.minutes"})
-      Timecop.freeze(Time.now.utc) do
-        expect(described_class.standard_capture_threshold(host)).to eq 4.minutes.ago.utc
-      end
-    end
+  describe ".perf_collect_all_metrics_queue" do
+    it "submits a single ems" do
+      expect_queued(
+        :class_name  => "Metric::Capture",
+        :method_name => "perf_collect_all_metrics",
+        :args        => [ems.id, "realtime", nil, nil, :exclude_storage => true],
+        :zone        => ems.zone_name
```

Comment:
👍  so this should be able to catch the `zone_name` vs `zone.id` issue

Reply:
provided that the test was written correctly in the first place :( - thanks for the catch